lib/entities/empty_state.rb

```diff
@@ -11,5 +12,9 @@ def initialize(empty_state, parent_empty_state=nil)
       @heading = parent_empty_state.heading || @heading
       @message = parent_empty_state.message || @message
     end
+
+    if parent_empty_state.nil?
+      @heading_tag = "h2"
+    end
```

Comment:
Instance variables ought to be immutable, so I'd turn `heading_tag` into a method that checks for `parent_empty_state`. I'd also make `parent_empty_state` an instance variable that isn't listed in `attr_reader` so that the `heading_tag` method has access to it but the outside world doesn't.

I don't think you should have to change your tests. Refactoring!